src/Events/Services/TraceLogService.cs

```
@@ -0,0 +1,117 @@
+                return cloudEvent.Id;
+            }
+            catch (Exception exception)
+            {
+                _logger.LogError(exception, "Error creating trace log entry with subscription details: {Message}", exception.Message);
+
+                // don't throw exception, we don't want to stop the event processing
+                return string.Empty;
+            }
+        }
+
+        /// <inheritdoc/>
+        public async Task<string> CreateWebhookResponseEntry(LogEntryDto logEntryDto)
+        {
+            var parseResult = Guid.TryParse(logEntryDto.CloudEventId, out Guid parsedGuid);
+
+            if (!parseResult)
+            {
+                _logger.LogError("Error creating trace log entry for webhook POST response: Invalid GUID");
+                return string.Empty;
+            }
+
+            if (string.IsNullOrEmpty(logEntryDto.CloudEventId) || logEntryDto.Endpoint == null || logEntryDto.StatusCode == null)
+            {
+                _logger.LogError("Error creating trace log entry for webhook POST response: Missing required input parameters");
+                return string.Empty;
+            }
+
+            try
+            {
+                var traceLogEntry = new TraceLog
+                {
+                    CloudEventId = parsedGuid,
+                    Resource = logEntryDto.CloudEventResource,
+                    EventType = logEntryDto.CloudEventType,
+                    Consumer = logEntryDto.Consumer,
+                    SubscriberEndpoint = logEntryDto.Endpoint.ToString(),
+                    SubscriptionId = logEntryDto.SubscriptionId,
+                    ResponseCode = (int?)logEntryDto.StatusCode,
+                    Activity = TraceLogActivity.WebhookPostResponse
+                };
+                await _traceLogRepository.CreateTraceLogEntry(traceLogEntry);
+                return logEntryDto.CloudEventId;
+            }
+            catch (Exception)
+            {
+                throw;
```

The other methods in this class are intentionally swallowing the exceptions from the repo. Should we do the same in here as well?